pkg/frontend/querymiddleware/codec.go

```diff
@@ -237,6 +237,7 @@ func (prometheusCodec) MergeResponse(responses ...Response) (Response, error) {
 
 	promResponses := make([]*PrometheusResponse, 0, len(responses))
 	promWarningsMap := make(map[string]struct{}, 0)
+	promInfosMap := make(map[string]struct{}, 0)
```

Comment:
it seems like there's no unit test for this or in fact promWarningsMap in codec_test.go